File: tests/fixtures/sample.java
public class SampleApp {
    private String name;
    private int counter;

    public SampleApp(String name) {
        this.name = name;
        this.counter = 0;
    }

    public int process(int value) {
        if (value > 0) {
            for (int i = 0; i < value; i++) {
                if (i % 2 == 0) {
                    counter += i;
                } else if (i % 3 == 0) {
                    counter += i * 2;
                }
            }
        }
        return counter;
    }

    public void resetCounter() {
        counter = 0;
        System.out.println("Counter reset for " + name);
    }

    public static void main(String[] args) {
        SampleApp app = new SampleApp("test");
        int result = app.process(10);
        System.out.println("Result: " + result);
    }
}

File: tests/fixtures/sample.cpp
#include <iostream>
#include <vector>
#include <string>

class DataProcessor {
private:
    std::vector<int> data;
    std::string name;

public:
    DataProcessor(const std::string& n) : name(n) {}

    void addData(int value) {
        data.push_back(value);
    }

    int processData() {
        int total = 0;
        for (size_t i = 0; i < data.size(); i++) {
            if (data[i] > 0) {
                total += data[i];
                if (data[i] > 100) {
                    total += static_cast<int>(data[i] * 0.1);
                }
            } else if (data[i] < 0) {
                total -= data[i] / 2;
            }
        }
        return total;
    }

    void clearData() {
        data.clear();
    }

    const std::string& getName() const {
        return name;
    }
};

int main() {
    DataProcessor processor("test");
    for (int i = 0; i < 10; i++) {
        processor.addData(i * 10);
    }
    int result = processor.processData();
    std::cout << "Result: " << result << std::endl;
    return 0;
}
